### DISCUSSION TITLE
Display messages/notifications

### DISCUSSION THREAD

**Discussion by qwertycandy** [netbox-community/netbox#10896] [Q&A]
Hello, I would like some help with messages, please. I'm writing a custom validator and I need to display a warning message if for example a device name doesn't fit the company's policy. I can't use the fail method - the edit request should be fulfilled, it's just supposed to warn the user as well. I would like to use a box message like in [#9650](https://github.com/netbox-community/netbox/discussions/9650), just with the warning level.

I tried something like this:
```python
from extras.validators import CustomValidator
from django.contrib import messages

class device_validator(CustomValidator):
	def validate(self, instance):
		if instance.name is not instance.name.upper():
			messages.warning(request, "Names of devices should be in all-caps.")		
			return
```

But clearly, my syntax is wrong - I get the "Server Error" window and following message:
```python
<class 'NameError'>

name 'request' is not defined
```
Could someone please give me an example how to display a message in this context?
Thank you.

**Comment by jeremystretch** [comment]
> I can't use the fail method - the edit request should be fulfilled, it's just supposed to warn the user as well.

The purpose of validation is to _prevent_ an erroneous or invalid action from taking place. There is no facility in place to permit an invalid action but provide a warning, as that would be of extremely limited utility. Instead, you might consider writing a [report](https://docs.netbox.dev/en/stable/customization/reports/) to routinely check for non-conforming names.

**Reply by qwertycandy** [reply]
Thank you - I get that this is a bit niche, but it would really help me to be able to use messages in reports. For example when user adds/edits a device that has no IP address assigned to it - I would like to display a warning message there, as perhaps they just forgot to assign it (and it that case a mistake could be caught immediatelly), but it could also be legitimate. 
Is there a way to refference/display messages within a validator, please, or is that technically impossible?

**Comment by kkthxbye-code** [comment]
Unrelated to your issue but this will not work:

```if instance.name is not instance.name.upper():```

The `is` keyword compares identity, you want to use `!=` instead which compares value.
